.github/workflows/auto_update_doc.yml

```diff
@@ -0,0 +1,45 @@
+name: Auto update documentation
+on:
+  pull_request_target:
+  workflow_dispatch:
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name == 'workflow_dispatch' }}
+    cancel-in-progress: true
+
+jobs:
+  auto-update-doc:
+    runs-on: ubuntu-latest
+
+    permissions:
+      # Give the default GITHUB_TOKEN write permission to commit and push the changed files back to the repository.
+      contents: write
+
+    steps:
+    - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      with:
+        repository: ${{ github.event.pull_request.head.repo.full_name }}
+        # Checkout the branch made in the fork. Will automatically push changes
+        # back to this branch.
+        ref: ${{ github.head_ref }}
+
+    - name: Setup Python
+      uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1 # v4.7.0
+    - name: Install ONNX from source and update documentation
+      run: |
+        python -m pip install -q --upgrade pip
+        python -m pip install -r requirements-release.txt
+        git submodule update --init --recursive
+        export ONNX_ML=1
+        pip install -e .
+        python onnx/defs/gen_doc.py
+        python onnx/gen_proto.py -l
+        python onnx/gen_proto.py -l --ml
+        python onnx/backend/test/stat_coverage.py
+        git diff -- . ':(exclude)onnx/onnx-data.proto' ':(exclude)onnx/onnx-data.proto3'
+
+    - name: Commit changes with updated documentation
```

Comment:
Can this create an infinite commit loop?

Reply:
It will only have one automatic commit when there is a new commit from the PR. Besides, I have set `cancel-in-progress: true` and so ideally there will be just one running CI at a time. Also this CI will only be triggered when there is a label "auto update doc", which can only be added by the maintainers.